### DISCUSSION TITLE
Queue: How to display which user added a song

### DISCUSSION THREAD

**Discussion by xenodirt** [eritislami/evobot#240] [General] [answered]
How do I make it so that my queue can identify which user added a song?
`message.author` doesn't work since it displays the user who requests for the queue list.

**Comment by brandonst** [comment, marked as answer]
Right now the song structure is only including the title, url, and duration, looks like this
```js
song = {
    title: songInfo.videoDetails.title,
    url: songInfo.videoDetails.video_url,
    duration: songInfo.videoDetails.lengthSeconds
}
```
You would need to manually input the author into it, for example:
```js
song = {
    title: songInfo.videoDetails.title,
    url: songInfo.videoDetails.video_url,
    duration: songInfo.videoDetails.lengthSeconds,
    user: message.author.tag
}
```
There are 2 places you would need to edit it, in `commands > play.js` and `commands > playlist.js`
Then, in this example, to access the person who added the song that's currently playing
You would do
```js
const serverQueue = message.client.queue.get(message.guild.id)
const author = serverQueue.songs[0].user
```

**Reply by aimankhairy** [reply]
where do i need to put the const serverQueue = message.client.queue.get(message.guild.id)
const author = serverQueue.songs[0].user ?

**Reply by brandonst** [reply]
you would put that in the command you're trying to use it in

**Comment by kasepta** [comment]
```
2020-10-25T09:05:40.630572+00:00 app[working.1]: /app/commands/play.js:17
2020-10-25T09:05:40.630599+00:00 app[working.1]:     const author = serverQueue.songs[0].user
2020-10-25T09:05:40.630600+00:00 app[working.1]:                                ^
2020-10-25T09:05:40.630601+00:00 app[working.1]: 
2020-10-25T09:05:40.630601+00:00 app[working.1]: TypeError: Cannot read property 'songs' of undefined
```
i have put the code to `commands > play.js` and `commands > playlist.js`
but get error log

**Reply by brandonst** [reply]
the error is happening because the server doesn't have a queue, meaning that the bot isn't playing any music, but why are you putting that in play.js? you were only supposed to put the song structure in those ones
